puffin/src/cli.rs

```diff
@@ -553,3 +563,29 @@ fn binary_attack<PB: ProtocolBehavior>(
     }
     Ok(())
 }
+
+fn check_if_puts_exist<PB: ProtocolBehavior>(
+    put_registry: &PutRegistry<PB>,
+    put_list: Vec<&str>,
+) -> bool {
+    let available_puts: Vec<&str> = put_registry.puts().map(|(name, _)| name).collect();
+
+    let non_available_puts: Vec<&str> = put_list
+        .iter()
+        .filter_map(|name| {
+            if available_puts.iter().any(|x| x == name) {
+                None
+            } else {
+                Some(*name)
+            }
+        })
+        .collect();
+
+    if non_available_puts.is_empty() {
+        true
+    } else {
+        println!("PUT not found : {}", non_available_puts.join(","));
+        println!("Available PUTs: {}", available_puts.join(","));
+        false
```

Comment:
Use logging functions instead of `println`.

Also, it's difficult to set a right log level here because the intent of the caller might vary. This suggests that here is not the correct place for these log functions.

How about doing a function `missing_puts` instead, returning the vec of missing PUTs? Then the intent is more clear and in the CLI you already perform a check anyway, which let you group your logging functions together with the `return ExitCode::FAILURE`. Then it's clear that the log level should be `error` in this case.